1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -1,47 +1,79 @@
 class Solution(object):
-	# Given an n x n binary matrix grid, we return the shortest clear path in 
-	# the matrix, or -1 for no clear path.
+# Given an n x n binary matrix grid, we return the shortest clear path in 
+# the matrix, or -1 for no clear path.
 
-	# A clear path is a path from the top-left cell to the bottom right cell
-	# such that:
+# A clear path is a path from the top-left cell to the bottom right cell
+# such that:
 
-	 	# 1. all visited cells are 0
-	 	# 2. all adjacent cells are different and share an edge or corner
+	# 1. all visited cells are 0
+	# 2. all adjacent cells are different and share an edge or corner
```

Comment:
What does "all adjacent cells are different" mean?

Reply:
Oh I am not sure actually...the original problem says:

"All the adjacent cells of the path are 8-directionally connected (i.e., they are different and they share an edge or a corner)."

...so I think I was copying that but I don't know what "they are different" means here.